src/Board.py

```diff
@@ -50,6 +51,85 @@ def __init__(self, style: str = "default", size: int = 5) -> None:
         else:
             raise ValueError("Unrecognised map style")
 
+    def get(self, coords):
+        return self.data[coords[1]][coords[0]]
+
+    def get_directly_adj(self, start=(0, 0), terain_filter=None) -> list:
+        adjacent_tile_list = []
+        if terain_filter is None:
+            if start[1] > 0:
+                if start[1] % 2 == 0:  # for even rows
+                    adjacent_tile_list.append((start[0], start[1] - 1))  # top left
+                    adjacent_tile_list.append((start[0] + 1, start[1] - 1))  # top right
+                else:  # for odd rows
+                    adjacent_tile_list.append((start[0] - 1, start[1] - 1))  # top left
+                    adjacent_tile_list.append((start[0], start[1] - 1))  # top right
+            if start[0] < BOARD_WIDTH - 2 or (
+                start[1] % 2 == 1 and start[0] < BOARD_WIDTH - 1
+            ):
+                adjacent_tile_list.append((start[0] + 1, start[1]))  # right
+            if start[1] < BOARD_HEIGHT - 1:
+                if start[1] % 2 == 0:  # for even rows
+                    adjacent_tile_list.append(
+                        (start[0] + 1, start[1] + 1)
+                    )  # bottom right
+                    adjacent_tile_list.append((start[0], start[1] + 1))  # bottom left
+                else:  # for odd rows
+                    adjacent_tile_list.append((start[0], start[1] + 1))  # bottom right
+                    adjacent_tile_list.append(
+                        (start[0] - 1, start[1] + 1)
+                    )  # bottom left
+            if start[0] > 0:
+                adjacent_tile_list.append((start[0] - 1, start[1]))  # left
+        else:
+            unfiltered_list = self.get_directly_adj(start, None)
+            for tile in unfiltered_list:
+                if self.get(tile)._terrain == terain_filter:
+                    adjacent_tile_list.append(tile)
+        return adjacent_tile_list  # = [TL,TR,R,BR,BL,L]
+
+    def get_indirectly_adj(
+        self, start=(0, 0), terain_filter=None, shipping_limit=0
+    ) -> list:
+        adjacent_tile_list = []
+        if self.get_directly_adj(start, Terrain.RIVER) == []:
+            return self.get_directly_adj(start, None)
+        if shipping_limit == 0:
+            direct_list = self.get_directly_adj(start, terain_filter)
+            for tile in direct_list:
+                if not self.found_tile(adjacent_tile_list, tile):
```

Comment:
I don't think we need a separate function for this.. in Python if you're comparing tuples like this you can just do:
```suggestion
                if tile not in adjacent_tile_list:
```